packages/evm/contracts/court/controller/Controlled.sol

```diff
@@ -88,58 +132,77 @@ contract Controlled is IsContract, ConfigConsumer {
 
     /**
     * @dev Internal function to fetch the controller's config governor
-    * @return Address of the controller's governor
+    * @return Address of the controller's config governor
     */
     function _configGovernor() internal view returns (address) {
         return controller.getConfigGovernor();
     }
 
     /**
-    * @dev Internal function to fetch the address of the DisputeManager module from the controller
+    * @dev Internal function to fetch the controller's modules governor
+    * @return Address of the controller's modules governor
+    */
+    function _modulesGovernor() internal view returns (address) {
+        return controller.getModulesGovernor();
+    }
+
+    /**
+    * @dev Internal function to fetch the address of the DisputeManager module
     * @return Address of the DisputeManager module
     */
     function _disputeManager() internal view returns (IDisputeManager) {
-        return IDisputeManager(controller.getDisputeManager());
+        return IDisputeManager(_getModuleCache(DISPUTE_MANAGER));
     }
 
     /**
-    * @dev Internal function to fetch the address of the Treasury module implementation from the controller
+    * @dev Internal function to fetch the address of the Treasury module implementation
     * @return Address of the Treasury module implementation
     */
     function _treasury() internal view returns (ITreasury) {
-        return ITreasury(controller.getTreasury());
+        return ITreasury(_getModuleCache(TREASURY));
     }
 
     /**
-    * @dev Internal function to fetch the address of the Voting module implementation from the controller
+    * @dev Internal function to fetch the address of the Voting module implementation
     * @return Address of the Voting module implementation
     */
     function _voting() internal view returns (ICRVoting) {
-        return ICRVoting(controller.getVoting());
+        return ICRVoting(_getModuleCache(VOTING));
     }
 
     /**
-    * @dev Internal function to fetch the address of the Voting module owner from the controller
+    * @dev Internal function to fetch the address of the Voting module owner
     * @return Address of the Voting module owner
     */
     function _votingOwner() internal view returns (ICRVotingOwner) {
-        return ICRVotingOwner(address(_disputeManager()));
+        return ICRVotingOwner(_getModuleCache(DISPUTE_MANAGER));
```

Comment:
I didn't catch this earlier, but I'm wondering if the concept of a "Voting owner" will be confusing given that we're allowed to switch modules around.

Would it be better to make this assumption only in the Voting module? (E.g. by using `_disputeManager()` and casting it to `ICRVotingOwner`)

Reply:
I liked that! Addressed in e078a12f1473cd1c439eeb7e3ee29746a03cde54